experimental/adaptive-dialog/docs/recognizers-rules-steps-reference.md

```diff
@@ -0,0 +1,739 @@
+    {
+        { "HelpIntent", "(?i)help" },
+        { "CancelIntent", "(?i)cancel|never mind" }
+    }
+};
+// Create an intent rule with the intent name
+var helpRule = new IntentRule("HelpIntent");
+
+// Create steps when the bookFlightRule triggers
+var helpSteps = new List<IDialog>();
+helpSteps.Add(new SendActivity("Hello, I'm the samples bot. At the moment, I respond to only help!"));
+helpRule.Steps = helpSteps;
+
+// Add the bookFlight rule to root dialog
+rootDialog.AddRule(helpRule);
+
+// Add a rule to capture unhandled intents. Unknown intent rule fires when a recognizedIntent event raised
+// by the recognizer is not handled by any rule added to the dialog.
+// Given the RegEx recognizer added to this dialog, this rule will fire when user says 'cancel'.
+// Although the recognizer returned 'cancel' intent, we have no rule attached to handled it. 
+// This rule will also fire when user says 'yo'. The recognizer will return a 'none' intent however that
+// intent is not caught by a rule added to this dialog. 
+var unhandledIntentRule = new UnknownIntentRule();
+var unhandledIntentSteps = new List<IDialog>();
+unhandledIntentSteps.Add(new SendActivity("Sorry, I did not recognize that"));
+unhandledIntentRule.Steps = unhandledIntentSteps;
+
+rootDialog.AddRule(unhandledIntentRule);
+```
+
+### Event rule
+As the name suggests, event rule enables you to catch and react to [events](#events) raised by any of the sub-system or your own dialog. Remember each step is a dialog and you can use the [EmitEvent](#Emit-event) step to emit a custom event. 
+
+``` C#
+// Create root dialog as an Adaptive dialog.
+var rootDialog = new AdaptiveDialog("rootDialog");
+rootDialog.AddRule(new EventRule([AdaptiveEvents.ActivityReceived], new List<IDialog>()
+{
+    new SendActivity("Event {turn.ActivityReceived.value} received")
+});
+```
+
+## Inputs 
+_Inputs_ are wrappers around [prompts][2] that you can use in an adaptive dialog step to ask and collect a piece of input from user, validate and accept it into memory. Inputs include these pre-built features - 
+- Accepts a property to bind to the new [state management][6] scopes. 
+- Performs existential check before prompting. 
+- Grounds input to the specified property if the input from user matches the type of entity expected. 
+- Accepts constraints - min, max, etc. 
+
+Adaptive dialogs support the following inputs - 
+- [TextInput](#TextInput)
+- [ChoiceInput](#ChoiceInput)
+- [ConfirmInput](#ConfirmInput)
+- [NumberInput](#NumberInput)
+
+### TextInput
+Use text input when you want to verbatim accept user input as value for a specific piece of information your bot is trying to collect. E.g. user's name, subject of an email etc. 
+
+**Note:** By default, TextInput will trigger a consultation before accepting user input. This is to allow a parent dialog to handle the specific user input in case their recognizer had a more definitive match via an intent or entity.
```

Comment:
You are not explaining what consultation is. Maybe refer and /or explain in details in the anatomy-of-adaptive?